state.go

```diff
@@ -140,15 +140,16 @@ type StateContext struct {
 	lastInventoryUpdateAttempt time.Time
 	lastAuthorizeAttempt       time.Time
 	fetchInstallAttempts       int
+	wakeupChan                 chan bool
```

Comment:
The wakeup channel is now global, and passed into the wait functions by reference, which should resolve the synchronization issues we've been facing.

Reply:
:+1: